packages/@lwc/engine-core/src/framework/stylesheet.ts

```diff
@@ -132,6 +132,12 @@ function evaluateStylesheetsContent(
                 // the stylesheet, while internally, we have a replacement for it.
                 stylesheet = getStyleOrSwappedStyle(stylesheet);
             }
+
+            // Check that this stylesheet was generated by our compiler
```

Comment:
Making this check here felt the most natural since it's where we go through and evaluate the stylesheets.

Reply:
One downside of this approach is that we are not verifying that the _array_ of stylesheets was generated by our compiler. But this is technically fine, as we have cases like `[...super.styleshesheets]` where users will be constructing their own arrays, and we allow it.

Reply:
> We allow it as in it's a supported use case?

Yes.

Overall I'm not concerned about validating the array; I just wanted to point out that we're not going to bother doing that. Validating just the function is fine.